### PR TITLE
Fix Button component, updated tests and added Jest Native

### PR DESCRIPTION
**Pull request by deelanM**

### CODE REVIEW

File: src/components/button/button.style.tsx [hunk 1]
```diff
@@ -45,7 +45,7 @@ export const getStyle = (theme: ThemeType, props: ButtonProps) => {
     ...createBorderRadiusStyles(props, theme.borderRadius),
     backgroundColor: getThemeColor(
       theme.colors,
-      props.bg || `${props.colorScheme}.500`
+      `${props.bg ?? props.colorScheme}.500`
```
Comment (yoannfleurydev) [inline src/components/button/button.style.tsx:48]: I think you don't want to do that. bg and colorScheme are two different things. If I put `gray.600` as a background I don't want the value to be `gray.600.500`

File: tsconfig.json [hunk 2]
```diff
@@ -22,7 +22,7 @@
     "skipLibCheck": true,
     "strict": true,
     "target": "esnext",
-    "types": ["react-native", "jest"]
+    "types": ["react-native", "jest", "@testing-library/jest-native"]
```
Comment (yoannfleurydev) [inline tsconfig.json:25]: Are you sure we need this ? It looks like this lib provides typings and suggests another way to use types

**Review by yoannfleurydev — CHANGES_REQUESTED** [review]

File: src/components/button/button.style.tsx (hunk 1 above)
Comment (deelanM) [reply src/components/button/button.style.tsx:48]: So what you're saying is we should expect Button bg prop to be called with a shade and colorScheme a color ? Like `<Button bg="red.500" colorScheme="blue">`  ?

File: tsconfig.json (hunk 2 above)
Comment (deelanM) [reply tsconfig.json:25]: Without it, the build action will fail https://github.com/BearStudio/react-native-ficus-ui/actions/runs/6362862113/job/17278149636
This is the only way I have found to fix it

File: src/components/button/button.style.tsx (hunk 1 above)
Comment (yoannfleurydev) [reply src/components/button/button.style.tsx:48]: I mean that final developer shouldn't be able to put `bg` and `colorScheme` (in fact they can, but some precedence rule will override the `colorScheme` with the `bg`. `bg` and `colorScheme` are two different things. `bg` is used to put precise values, `colorScheme` is used to provide a color and then the component will adapt itself with it using the [0, ..., 900] palette.

File: src/theme/theme.service.ts [hunk 3]
```diff
@@ -129,38 +129,34 @@ export const getThemeColor = (
...
 ): string => {
-  if (themeColors && value) {
-    if (
-      themeColors.hasOwnProperty(value) &&
-      typeof themeColors[value] !== 'undefined'
-    ) {
-      const colorValue: string | String = themeColors[value] as string;
-      return colorValue as string;
-    }
+  if (!(themeColors && value) || value.startsWith('#')) {
```
Comment (ntorionbearstudio) [inline src/theme/theme.service.ts:132]: not right way to do (search by '#' symbol)

**Pull request merged**